sdk/core/core-amqp/src/errors.ts

```diff
@@ -548,46 +568,43 @@ function isBrowserWebsocketError(err: any): boolean {
 }
 
 /**
- * Translates the AQMP error received at the protocol layer or a generic Error into a MessagingError.
+ * Translates the AQMP error received at the protocol layer or a SystemError into a MessagingError.
+ * All other errors are returned unaltered.
  *
  * @param {AmqpError} err The amqp error that was received.
  * @returns {MessagingError} MessagingError object.
  */
-export function translate(err: AmqpError | Error): MessagingError {
+export function translate(err: AmqpError | Error): MessagingError | Error {
```

Comment:
Since the return type can be `Error` as well, wouldnt the instances of `if (translatedError.retryable)` checks made after `const translatedError = translate(err)` fail to compile as `retryable` doesnt exist on `MessagingError | Error` ?

Reply:
Hmmm, I was trying to keep non SystemError/MessagingErrors unaltered, in which case they would most likely not have `translated` or `retryable` properties defined.

I think we can either add those fields to the passed in error if they don't exist and either change the return type back to `MessagingError` or add an intersection type to `Error`:
```suggestion
export function translate(err: AmqpError | Error): MessagingError | (Error & { translated: boolean; retryable: boolean}) {
```

Making the return type `MessagingError` seems simpler, but I'm worried that this implies that the user will always get back an instance of `MessagingError` (so `translate(error) instanceof MessagingError === true` should be true), when this is no longer the case.

@bterlson Since these changes started because we wanted `error.name` to reflect the class, is the point I raise here also relevant?

Reply:
I don't like the intersection type since the ultimate effect of the intersection is a type that just looks like MessagingError but declared in a different way (if I understand correctly). If that's true, better to just return MessagingError here.

That said, it is not good practice to assert the type of an error you caught without checking it, generally speaking. Even if we did the work to translate some errors, it's still possible via bugs or other means to get a regular old JS error out of these APIs or at least in your catch block, and those wouldn't have the properties they'd expect either. In other words, `MessagingError | Error` is exactly the type that users have to grapple with in their catch block anyway.

THAT said, @chradek I'm not sure what you're asking, can you clarify? 😀

Reply:
You essentially answered what I was trying to ask: Should we claim we return a `MessagingError` always, or return `Error` as well? (Which I believe you're saying we should return both)

I get your point about not wanting to use an intersection type and agree. In that case it seems better to update the code in event hubs that's impacted by this to handle the fact that some properties may be missing.